osbenchmark/builder/installers/docker_installer.py

```diff
@@ -0,0 +1,166 @@
+        self.provision_config_instance = provision_config_instance
+
+    def install(self, host, binaries):
+        node = self._create_node()
+        self._prepare_node(host, node)
+
+        return node
+
+    def _create_node(self):
+        node_name = str(uuid.uuid4())
+        node_port = int(self.provision_config_instance.variables["node"]["port"])
+        node_root_dir = os.path.join(self.provision_config_instance.variables["test_execution_root"], node_name)
+        node_data_paths = [os.path.join(node_root_dir, "data", str(uuid.uuid4()))]
+        node_binary_path = os.path.join(node_root_dir, "install")
+
+        return Node(name=node_name,
+                    port=node_port,
+                    pid=None,
+                    root_dir=node_root_dir,
+                    binary_path=node_binary_path,
+                    data_paths=node_data_paths,
+                    telemetry=None)
+
+    def _prepare_node(self, host, node):
+        node_log_dir = os.path.join(node.root_dir, "logs", "server")
+        node_heap_dump_dir = os.path.join(node.root_dir, "heapdump")
+
+        directories_to_create = [node.binary_path, node_log_dir, node_heap_dump_dir, node.data_paths[0]]
+        for directory_to_create in directories_to_create:
+            self._create_directory(host, directory_to_create)
+
+        mounts = self._prepare_mounts(host, node)
+        docker_cfg = self._render_template_from_docker_file(self._get_docker_vars(node, node_log_dir,
+                                                                                  node_heap_dump_dir, mounts))
+        self.logger.info("Installing Docker container with configuration:\n%s", docker_cfg)
+
+        docker_compose_file = os.path.join(node.binary_path, "docker-compose.yml")
+        with open(docker_compose_file, mode="wt", encoding="utf-8") as f:
+            f.write(docker_cfg)
+        self.executor.execute(host, "cp {0} {0}".format(docker_compose_file))
+
+    def _create_directory(self, host, directory):
+        # Create directory locally and on the host
+        io.ensure_dir(directory)
+        self.executor.execute(host, "mkdir -m 0777 -p " + directory)
+
+    def _prepare_mounts(self, host, node):
+        mounts = {}
+        for provision_config_instance_config_path in self.provision_config_instance.config_paths:
+            mounts.update(self._prepare_mounts_from_config_path(host, node, provision_config_instance_config_path))
+
+        return mounts
+
+    def _prepare_mounts_from_config_path(self, host, node, config_path):
+        config_path_mounts = {}
+        for root, _, files in os.walk(config_path):
+            config_path_mounts.update(self._prepare_mounts_from_config_path_root(host, node, config_path, root, files))
+
+        return config_path_mounts
+
+    def _prepare_mounts_from_config_path_root(self, host, node, config_path, root, files):
+        env = jinja2.Environment(loader=jinja2.FileSystemLoader(root),
```

Comment:
Not required for this PR but I think we might want to move this Jinja environment setup into its own utils function. I think it's used several times across the codebase and ensuring that each caller passes the autoescape characters seems like a pain.